vmc/resource_vmc_sddc.go

```diff
@@ -165,15 +177,52 @@ func resourceSddc() *schema.Resource {
 				Computed: true,
 			},
 		},
+		CustomizeDiff: func(d *schema.ResourceDiff, meta interface{}) error {
+
+			newInstanceType := d.Get("host_instance_type").(string)
+
+			switch newInstanceType {
+
+			case "I3_METAL":
+
+				if d.Get("storage_capacity").(string) != "" {
+
+					return fmt.Errorf("storage_capacity is not supported for host_instance_type %q", newInstanceType)
+
+				}
+			case "R5_METAL":
+
+				if d.Get("storage_capacity").(string) == "" {
+
+					return fmt.Errorf("storage_capacity is required for host_instance_type %q", newInstanceType)
+
+				}
+
+			default:
+
+				// All other types support storage_capacity,
+
+				// so there is nothing to check.
+
+				return nil
+
+			}
+			return nil
+		},
 	}
 }
 
 func resourceSddcCreate(d *schema.ResourceData, m interface{}) error {
+	var storageCapacityConverted int64
```

Comment:
curious.. why is this int64?

Reply:
The AwsSddcConfig struct defined in the Go SDK expects the value for storage capacity to be int64 similar to numHosts which is also expected to be int64.